crates/re_viewer/src/ui/selection_panel.rs

```diff
@@ -409,32 +411,50 @@ fn entity_props_ui(
+        ColorMapper::Colormap(Colormap::Grayscale) => re_renderer::Colormap::Grayscale,
+        ColorMapper::Colormap(Colormap::Turbo) => re_renderer::Colormap::Turbo,
+        ColorMapper::Colormap(Colormap::Viridis) => re_renderer::Colormap::Viridis,
+        ColorMapper::Colormap(Colormap::Plasma) => re_renderer::Colormap::Plasma,
+        ColorMapper::Colormap(Colormap::Magma) => re_renderer::Colormap::Magma,
+        ColorMapper::Colormap(Colormap::Inferno) => re_renderer::Colormap::Inferno,
+    };
 
     ui.label("Color map");
-    egui::ComboBox::from_id_source("color_mapper")
+    egui::ComboBox::from_id_source("color map select")
         .selected_text(current.to_string())
         .show_ui(ui, |ui| {
             ui.style_mut().wrap = Some(false);
-            ui.set_min_width(64.0);
 
-            // TODO(cmc): that is not ideal but I don't want to import yet another proc-macro...
-            let mut add_label = |proposed| {
-                if ui
-                    .selectable_label(current == proposed, proposed.to_string())
-                    .clicked()
-                {
-                    entity_props.color_mapper = EditableAutoValue::UserEdited(proposed);
-                }
-            };
-
-            add_label(ColorMapper::Colormap(Colormap::Grayscale));
-            add_label(ColorMapper::Colormap(Colormap::Turbo));
-            add_label(ColorMapper::Colormap(Colormap::Viridis));
-            add_label(ColorMapper::Colormap(Colormap::Plasma));
-            add_label(ColorMapper::Colormap(Colormap::Magma));
-            add_label(ColorMapper::Colormap(Colormap::Inferno));
+            egui::Grid::new("colormap_selector")
+                .num_columns(2)
+                .show(ui, |ui| {
+                    for option in re_renderer::Colormap::ALL {
+                        let mut value = current;
+                        if ui
+                            .selectable_value(&mut value, option, option.to_string())
+                            .changed()
+                        {
+                            let value = match value {
+                                re_renderer::Colormap::Grayscale => Colormap::Grayscale,
+                                re_renderer::Colormap::Turbo => Colormap::Turbo,
+                                re_renderer::Colormap::Viridis => Colormap::Viridis,
+                                re_renderer::Colormap::Plasma => Colormap::Plasma,
+                                re_renderer::Colormap::Magma => Colormap::Magma,
+                                re_renderer::Colormap::Inferno => Colormap::Inferno,
+                            };
+                            entity_props.color_mapper =
+                                EditableAutoValue::UserEdited(ColorMapper::Colormap(value));
+                        }
+                        colormap_preview_ui(ctx.render_ctx, ui, option);
+                        ui.end_row();
+                    }
+                });
```

Comment:
it looks like you copied the code from `ColorMapping::ui` to here? Why not create a reusable `fn colormap_dropdown_button_ui` instead? It can return the new selection (if any) so that at the call-site it can be assigned to an `EditableAutoValue` as needed here.

It will make it much easier to add the same code to a third place, and will also reduce the chance of copy-paste errors (like having `ui.style_mut().wrap = Some(false);` in only one of the dropdowns).

The essential question should be this: should our colormap select dropdown look the same? Then they should be the same function